ansible/roles/create-ecr-build-role/vars/main.yml

```diff
@@ -0,0 +1,82 @@
+
+build_role_name: "build-{{ service_id }}"
+aws_ecs_policy_name: ecs-build-policy
+assume_role_policy:
+  Version: "2012-10-17"
+  Statement:
+    - Sid: ""
+      Effect: Allow
+      Principal:
+        AWS: "arn:aws:iam::{{ aws_account_id }}:role/agents-ecs-task"
+      Action: "sts:AssumeRole"
+aws_ecs_policy:
+  Version: "2012-10-17"
+  Statement:
+    - Sid: ecrrepo
+      Effect: Allow
+      Action:
+        - "ecr:CreateRepository"
+        - "ecr:TagResource"
+      Resource: [
+        "arn:aws:ecr:{{ aws_region }}:{{ aws_account_id }}:repository/{{ service_id }}_*"
+      ]
+      Condition:
+        StringEquals:
+          aws:RequestTag/api-service: "{{ service_id }}"
+    - Sid: ecr
+      Effect: Allow
+      Action:
+        - "ecr:BatchCheckLayerAvailability"
+        - "ecr:BatchGetImage"
+        - "ecr:DescribeImages"
+        - "ecr:DescribeImageScanFindings"
+        - "ecr:GetDownloadUrlForLayer"
+        - "ecr:GetLifecyclePolicy"
+        - "ecr:GetLifecyclePolicyPreview"
+        - "ecr:GetRepositoryPolicy"
+        - "ecr:ListTagsForResource"
+        - "ecr:CompleteLayerUpload"
+        - "ecr:InitiateLayerUpload"
+        - "ecr:PutImage"
+        - "ecr:PutImageScanningConfiguration"
+        - "ecr:PutImageTagMutability"
+        - "ecr:StartImageScan"
+        - "ecr:StartLifecyclePolicyPreview"
+        - "ecr:UploadLayerPart"
+      Resource: [
+        "arn:aws:ecr:{{ aws_region }}:{{ aws_account_id }}:repository/{{ service_id }}_*"
+      ]
+    - Sid: ecrdesc
+      Effect: Allow
+      Action:
+        - "ecr:DescribeRepositories"
+        - "ecr:GetAuthorizationToken"
+      Resource: "*"
+    - Sid: ssmget
+      Effect: Allow
+      Action:
+        - "ssm:GetParameter"
+        - "ssm:GetParameters"
+        - "ssm:GetParametersByPath"
+      Resource:
+        - "arn:aws:ssm:{{ aws_region }}:{{ aws_account_id }}:parameter/{{ account }}/platform-common/*"
+        - "arn:aws:ssm:{{ aws_region }}:{{ aws_account_id }}:parameter/{{ account }}/api-deployment/{{ service_id }}/*"
+    - Sid: ssmput
+      Effect: Allow
+      Action:
+        - "ssm:PutParameter"
+      Resource:
+        - "arn:aws:ssm:{{ aws_region }}:{{ aws_account_id }}:parameter/{{ account }}/api-deployment/{{ service_id }}/*"
+    - Sid: secretsmanagerget1
+      Effect: Allow
+      Action:
+        - "secretsmanager:GetSecretValue"
+      Resource: "arn:aws:secretsmanager:{{ aws_region }}:{{ aws_account_id }}:secret:{{ account }}/*"
+      Condition:
+        ForAnyValue:StringLike:
+          secretsmanager:ResourceTag/AllowedServices1:
+            - "{{ service_id }}"
+            - "{{ service_id }} *"
+            - "* {{ service_id }} *"
+            - "* {{ service_id }}"
+            - "all"
```

Comment:
why the above if it just gets access to `all`?

Reply:
ah this is the other way around .. this is allowed to access things that are tagged with 'all' .. ( clone of the existing )